File: exercism/java/rna-transcription/src/main/java/RnaTranscription.java
class RnaTranscription {

    String transcribe(String dnaStrand) {
        char[] splittedDna = dnaStrand.toCharArray();

        for(int i = 0; i < splittedDna.length; i++) {
            switch(splittedDna[i]) {
                case 'C':
                    splittedDna[i] = 'G';
                    break;
                case 'T':
                    splittedDna[i] = 'A';
                    break;
                case 'G':
                    splittedDna[i] = 'C';
                    break;
                case 'A':
                    splittedDna[i] = 'U';
                    break;
                default:
                    throw new IllegalArgumentException("Invalid input");
            }
        }
        return new String(splittedDna);
    }

}

File: exercism/typescript/rna-transcription/rna-transcription.ts
export function toRna(dnaStrand: string) {
  const translations = new Map();
  translations.set("G", "C");
  translations.set("C", "G");
  translations.set("T", "A");
  translations.set("A", "U");

  // let rnaStrand: string[] = [];
  // for (let i = 0; i < dnaStrand.length; i++) {
  //   if (!translations.has(dnaStrand.at(i))) {
  //     throw new Error("Invalid input DNA.");
  //   }
  //   rnaStrand.push(translations.get(dnaStrand.at(i)));
  // }
  // return rnaStrand.join("");

  return [...dnaStrand].map(nucleotides => {
    if (!translations.has(nucleotides)) {
      throw new Error("Invalid input DNA.");
    }
    return translations.get(nucleotides);
  }).join("");
}